R/base.R

```diff
@@ -53,17 +53,85 @@ str.R7_base_class <- function(object, ..., nest.lev = 0) {
   print(object, ..., nest.lev = nest.lev)
 }
 
-base_classes <- list(
-  logical = new_base_class("logical"),
-  integer = new_base_class("integer"),
-  double = new_base_class("double"),
-  complex = new_base_class("complex"),
-  character = new_base_class("character"),
-  raw = new_base_class("raw"),
+#' Base classes
+#'
+#' @description
+#' These classes represent base types allowing them to be used within R7.
+#' Most correspond directly to the obvious base type. There are three
+#' exceptions:
+#'
+#' * `class_numeric` is a union of `class_integer` and `class_double`.
+#' * `class_atomic` is a union of `class_logical`, `class_numeric`,
+#'   `class_complex`, and `class_raw`.
+#' * `class_vector` is a union of `class_atomic`, `class_list`, and
+#'   `class_expression`.
```

Comment:
I had a random thought that `class_numeric` and friends could be `union_numeric` to be more self documenting, but that might not be a great idea

Reply:
I think it's nice to have a common prefix here; and the prefix really means "some wrapper around a base class/type".